torch/csrc/distributed/c10d/reducer.cpp

```diff
@@ -938,7 +938,7 @@ void Reducer::runGradCallbackForVariable(
     GradCallback&& cb) {
   auto context_ptr = rpc_context_.context_ptr.load();
   if (context_ptr == nullptr) {
-    cb(variable.grad());
+    cb(variable.mutable_grad());
```

Comment:
This one also looks very suspicious!

Reply:
All the calls in the reducer are protected by a single mutex at the top level so this should be fine.